powerauth-restful-security-base/src/main/java/io/getlime/security/powerauth/rest/api/base/provider/PowerAuthEncryptionProviderBase.java

```diff
@@ -111,15 +111,20 @@
             final String ephemeralPublicKey = eciesRequest.getEphemeralPublicKey();
             final String encryptedData = eciesRequest.getEncryptedData();
             final String mac = eciesRequest.getMac();
+            final String nonce = eciesRequest.getNonce();
 
-            // Verify ECIES request data
+            // Verify ECIES request data. Nonce is required for protocol 3.1+
             if (ephemeralPublicKey == null || encryptedData == null || mac == null) {
                 throw new PowerAuthEncryptionException("Invalid ECIES request data");
             }
+            if (nonce == null && !"3.0".equals(encryptionContext.getVersion())) {
+                throw new PowerAuthEncryptionException("Missing nonce in ECIES request data");
```

Comment:
Missing logging.

Reply:
I want to be consistent with the rest of the code. There's no logging just a few lines above in the same situation.

Reply:
Ok